addons/mail/static/src/messaging/entity/thread_viewer/thread_viewer.js

```diff
@@ -191,14 +218,36 @@ function ThreadViewerFactory({ Entity }) {
             });
 
             if (thread && this.thread !== thread) {
+                clearTimeout(this._loaderTimeout);
                 this.link({ thread });
                 if (!this.threadCache.isLoaded && !this.threadCache.isLoading) {
                     this.threadCache.loadMessages();
                 }
             }
             if (this.threadCache !== prevThreadCache) {
+                if (this._loaderTimeout) {
+                    clearTimeout(this._loaderTimeout);
+                }
                 this._writeComponentHint('change-of-thread-cache');
             }
+
+            if(!this.thread) {
+                this._write({ isShowingLoading:false });
+                this._loaderTimeout = null;
```

Comment:
move it closer to clearTimeouts. Maybe define a private method that simply make clearTimeout + setting to falsy value